src/UseCases/HandlePayPalPaymentNotification/HandlePayPalPaymentNotificationUseCase.php

```diff
@@ -190,4 +198,44 @@ private function createChildDonation( Donation $donation, PayPalNotificationRequ
 		}
 		return $childDonation;
 	}
+
+	private function newDonorFromRequest( PayPalNotificationRequest $request ): Donor {
+		return new Donor(
+			$this->newPersonNameFromRequest( $request ),
+			$this->newPhysicalAddressFromRequest( $request ),
+			$request->getPayerEmail()
+		);
+	}
+
+	private function newPersonNameFromRequest( PayPalNotificationRequest $request ): PersonName {
+		$name = PersonName::newPrivatePersonName();
+		$name->setFirstName( $request->getPayerFirstName() );
+		$name->setLastName( $request->getPayerLastName() );
+		$name->freeze();
```

Comment:
You mean to catch the unlikely case that a `null` value comes from the request? Is that even possible withe all the type checking? 

Or should I put `->assertNoNullFields` in to guard against some change in the future where the internal values of newly instantiated person objects are no longer empty strings and `null` values instead?

I's a bit reluctant to do it since that means another try/catch block ...